xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/macros.vm

```diff
@@ -387,18 +387,20 @@ $xwiki.getUserName("xwiki:${username}")
   #end
   #if(!$picturelist.containsKey("${username}--${size}"))
     #getUserAvatarURL($username $avatarURL $size)
+    #set ($resolvedUser = $services.model.resolveDocument($username))
+    #set ($userNameEscaped = $escapetool.xml($escapetool.html($xwiki.getPlainUserName($resolvedUser))))
```

Comment:
why escaping both xml and html here? that would need a comment I think.

Reply:
And `getPlainUserName` is not supposed to perform that escape?

Reply:
Since xml and html escaping are pretty much the same thing it means you are doing double escaping here which I really doubt is the intent.

Reply:
> And getPlainUserName is not supposed to perform that escape?

No, getPlainUserName return plain text and not HTML so it's not escaped for HTML.

Reply:
> No, getPlainUserName return plain text and not HTML so it's not escaped for HTML.

But don't we want it to do the escape?

Reply:
> But don't we want it to do the escape?

Again it return plain text, escaping means returning HTML since the escaping syntax is HTML syntax. This value is used in various contextes which includes not HTML contextes.

Reply:
The double escaping might not be the most elegant solution but is needed for one use case related to the use of the username in the alt attribute of the user profile icon, used in the comments and in the top right header. In this case a simple html escaping is not enough to fix the issue.

Reply:
`alt` does not interpret the HTML as far as I can see so it should not need to be escaped twice (only once as any XML inserted in an attribute value).